### PR TITLE
perl-sys-cpu

### PR DESCRIPTION
**Pull request by jupollet**

### CODE REVIEW

**Comment by jupollet** [comment]
@conda-forge/staged-recipes 

Hi,
can you merge this pull requests please .

File: recipes/perl-sys-cpu/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    string: pl522_0
+    skip: true  # [win]  
+    skip: true  # [osx]
+requirements:
+    host:
+        - binutils_impl_linux-64 2.28.1 h04c84fa_2
+        - binutils_linux-64 7.2.0 hc67e822_10
+        - gcc_impl_linux-64 7.2.0 hc5ce805_2
+        - gcc_linux-64 7.2.0 h196ecd4_10
+        - libgcc-ng 7.2.0 h7cc24e2_2
```
Comment (msarahan) [inline recipes/perl-sys-cpu/meta.yaml]: hard-coding the compiler like this is not advisable.  You should be using the compiler jinja2 expressions.  Please read https://conda.io/docs/user-guide/tasks/build-packages/compiler-tools.html#anaconda-compilers-and-conda-build-3

Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: ok thanks, i have take an existing recipe in anaconda.org

Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: It's done can we merge now ?

File: recipes/perl-sys-cpu/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    skip: true  # [osx]
+requirements:
+    host:
+        - {{ compiler('c') }}
+
+        #- perl-threaded >=5.22.0
+        - perl >=5.22.0
+        #- perl-extutils-makemaker 7.24
+    run:
+        - libgcc-ng 7.2.0.*
```
[the remark this answers is not in the document]
Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: Sorry but, what exact lines i have to remove please ?

[the remark this answers is not in the document]
Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: I think it's good now

File: recipes/perl-sys-cpu/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+package:
+    name: {{ name }}
+    version: {{ version }}
+source:
+    fn: Sys-CPU-0.61.tar.gz
+    url: https://cpan.metacpan.org/authors/id/M/MZ/MZSANFORD/Sys-CPU-0.61.tar.gz
+    sha256: {{ hash }}
+build:
+    number: '0'
+    string: pl522_0
```
Comment (msarahan) [inline recipes/perl-sys-cpu/meta.yaml]: You should probably not set this yourself.  Conda-build has an automatic setting that is usually good to use.  I don't think this should prevent a merge here, necessarily, it's just not good practice in general.

Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: I see this in many recipe . What part of this recipe should i remove please ?

Comment (msarahan) [reply recipes/perl-sys-cpu/meta.yaml]: just remove the string line completely.  You are probably looking at the recipe in the package, which is a fully-rendered form that has many things baked in.  You should be referencing other raw recipes, such as the ones on conda-forge.  It is very unusual for them to set build/string.

Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: Ok, thanks

Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: It's ok now.

File: recipes/perl-sys-cpu/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# This file created by conda-build 3.0.15+197.g41d59b5
+# meta.yaml template originally from:
+# /home/rdonnelly/conda/staged-recipes/recipes/perl-sys-cpu/0.61, last modified Sat Nov 11 23:55:48 2017
+# ------------------------------------------------
+#downgrade for perl 5.22 june 4th 2018
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Remove the comments above

File: recipes/perl-sys-cpu/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+{% set name = "perl-sys-cpu" %}
+{% set version = "0.61" %}
+{% set hash = "250a86b79c231001c4ae71d2f66428092a4fbb2070971acafd471aa49739c9e4" %}
+ 
+package:
+    name: {{ name }}
+    version: {{ version }}
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Add space after this line

File: recipes/perl-sys-cpu/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+source:
+    fn: Sys-CPU-0.61.tar.gz
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Remove this line

File: recipes/perl-sys-cpu/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+    url: https://cpan.metacpan.org/authors/id/M/MZ/MZSANFORD/Sys-CPU-0.61.tar.gz
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Change `0.61` here to `{{ version }}`

File: recipes/perl-sys-cpu/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    sha256: {{ hash }}
+build:
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Add space before each of these main section

File: recipes/perl-sys-cpu/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+    number: '0'
+    skip: true  # [win]  
+    skip: true  # [osx]
+requirements:
+    host:
+        - {{ compiler('c') }}
+
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Remove this empty line

File: recipes/perl-sys-cpu/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+test:
+    imports:
+        - Sys::CPU
+about:
+    home: http://metacpan.org/pod/Sys::CPU
+    license: perl5
+    summary: Perl extension for getting CPU information. Currently only number of
+        CPU's supported.
+extra:
+
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Remove this empty line

File: recipes/perl-sys-cpu/meta.yaml [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    skip: true  # [osx]
+requirements:
+    host:
+        - {{ compiler('c') }}
+
+        #- perl-threaded >=5.22.0
+        - perl >=5.22.0
+        #- perl-extutils-makemaker 7.24
+    run:
+
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Remove this empty line

File: recipes/perl-sys-cpu/meta.yaml [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+    sha256: {{ hash }}
+build:
+    number: '0'
+    skip: true  # [win]  
+    skip: true  # [osx]
+requirements:
+    host:
+        - {{ compiler('c') }}
+
+        #- perl-threaded >=5.22.0
```
Comment (isuruf) [inline recipes/perl-sys-cpu/meta.yaml]: Remove these commented out lines

Comment (jupollet) [reply recipes/perl-sys-cpu/meta.yaml]: all the changes are done on perl-sys-cpu-feedstock

**Pull request merged**